### PR TITLE
[POAE7-2647] support STRUCT result for groupby agg for arrow format

### PR DESCRIPTION
**Pull request by imajiayu**
### What changes were proposed in this pull request?
support STRUCT result for groupby agg for arrow format.
Specifically handle groupby Partial AVG in fetch result stage.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
CiderAggArrowTest.cpp

### CODE REVIEW

File: cider/exec/module/CiderRuntimeModule.cpp [hunk 1]
```diff
@@ -888,6 +889,40 @@ CiderBatch CiderRuntimeModule::setSchemaAndUpdateAggResIfNeed(
         }
       }
     }
+  } else if (ciderCompilationOption_.use_cider_data_format) {
```
Comment (jikunshang) [inline cider/exec/module/CiderRuntimeModule.cpp:892]: if (ciderCompilationOption_.use_cider_data_format && is_group_by_) would be better?

File: cider/include/cider/batch/CiderBatch.h [hunk 2]
```diff
@@ -115,6 +115,14 @@ class CiderBatch {
 
   std::string toStringForArrow() const;
 
+  ArrowSchema* getArrowSchema() const { return arrow_schema_; }
+
+  ArrowArray* getArrowArray() const { return arrow_array_; }
+
+  ArrowSchema* setArrowSchema(ArrowSchema* schema) { arrow_schema_ = schema; }
```
Comment (jikunshang) [inline cider/include/cider/batch/CiderBatch.h:122]: I am afraid this method may be unsafe.

Comment (imajiayu) [reply cider/include/cider/batch/CiderBatch.h:122]: I will use CiderBatch Constructor

File: cider/include/cider/batch/CiderBatch.h [hunk 3]
```diff
@@ -115,6 +115,12 @@ class CiderBatch {
 
   std::string toStringForArrow() const;
 
+  ArrowSchema* getArrowSchema() const { return arrow_schema_; }
+
+  ArrowArray* getArrowArray() const { return arrow_array_; }
+
+  void removeOwnerShip() { this->ownership_ = false; }
```
Comment (jikunshang) [inline cider/include/cider/batch/CiderBatch.h:122]: add some comments here?

**Review by jikunshang — DISMISSED** [review]

File: cider/include/cider/batch/CiderBatch.h [hunk 4]
```diff
@@ -115,6 +115,14 @@ class CiderBatch {
 
   std::string toStringForArrow() const;
 
+  ArrowSchema* getArrowSchema() const { return arrow_schema_; }
+
+  ArrowArray* getArrowArray() const { return arrow_array_; }
```
Comment (harborn) [inline cider/include/cider/batch/CiderBatch.h:120]: modify same

File: cider/tests/functionality/CiderAggArrowTest.cpp [hunk 5]
```diff
@@ -1019,27 +1019,53 @@ TEST_F(CiderPartialAVGFpArrowTest, mixedGroupbyPartialAVG) {
   std::vector<int64_t> expect_count3({0, 1, 1});
   std::vector<int64_t> expect_count4({0, 0, 0});
 
-  ArrowSchema* expect_schema;
-  ArrowArray* expect_array;
+  ArrowSchema** expect_schemas = (ArrowSchema**)malloc(4 * sizeof(ArrowSchema*));
+  ArrowArray** expect_arrays = (ArrowArray**)malloc(4 * sizeof(ArrowArray*));
```
Comment (harborn) [inline cider/tests/functionality/CiderAggArrowTest.cpp:1023]: maybe this two variable cause memory leak?
emm, this is tests codes, maybe not consider too much...

File: cider/exec/module/CiderRuntimeModule.cpp [hunk 6]
```diff
@@ -34,6 +34,7 @@
 #include "exec/plan/parser/TypeUtils.h"
 #include "exec/template/CountDistinct.h"
 #include "exec/template/OutputBufferInitialization.h"
+#include "tests/utils/ArrowArrayBuilder.h"
```
[the remark this answers is not in the document]
Comment (imajiayu) [reply cider/exec/module/CiderRuntimeModule.cpp:37]: This builder is also used in 'CiderBatchUtils.cpp' for non-groupby agg, maybe we should move it out from tests/utils.

File: cider/include/cider/batch/CiderBatch.h [hunk 7]
```diff
@@ -115,6 +115,14 @@ class CiderBatch {
 
   std::string toStringForArrow() const;
 
+  ArrowSchema* getArrowSchema() const { return arrow_schema_; }
```
[the remark this answers is not in the document]
Comment (imajiayu) [reply cider/include/cider/batch/CiderBatch.h:118]: there are a lot of codes need to modify if we the change the return type from ptr to class

File: cider/exec/module/CiderRuntimeModule.cpp (hunk 6 above)
[the remark this answers is not in the document]
Comment (yma11) [reply cider/exec/module/CiderRuntimeModule.cpp:37]: Let's check out whether we still need such logic in nextgen. If not, for arrow support, let's keep it like this.

[the remark this answers is not in the document]
Comment (yma11) [reply cider/exec/module/CiderRuntimeModule.cpp:37]: Confirmed with @bigPYJ1151 offline that we will can generate struct array during codegen directly and current logic won't be needed. So let's keep this for the time being.

**Review by yma11 — APPROVED** [review]

**Pull request merged**